packages/roosterjs-editor-plugins/lib/plugins/CustomReplace/CustomReplace.ts

```diff
@@ -131,9 +145,9 @@ export default class CustomReplacePlugin implements EditorPlugin {
             const [sourceMatch, replacementMatch] = replacement.matchSourceCaseSensitive
                 ? [stringToSearch, replacement.sourceString]
                 : [lowerCaseStringToSearch, replacement.sourceString.toLocaleLowerCase()];
-
+            const cleanSourceMatch = sourceMatch.replace(/\s/g, ' ');
```

Comment:
reg exp replacing is an expensive operation, so we should move it out of the loop.
For example:
const originalStringToSearch = stringToSearch.replace(/s/g, ' ');
const lowerCaseStringToSearch = originalStringToSearch .toLocalLowerCase();
for (... ) {
   const [sourceMatch, replacementMatch] = replacement.matchSourceCaseSensitive ? [originalStringToSearch , replacement.sourceString] : [lowerCaseStringToSearch , replacement.sourceString.toLocaleLowerCase()];